frontend/src/views/supp-allowances/SupplementaryFormView.vue

```diff
@@ -183,6 +143,59 @@ export default {
         this.setFailureAlert('Failed to load supplementary applications')
       }
     },
+    async saveApplication(showAlert = false) {
+      try {
+        this.loading = true
+        this.$emit('process', true)
+        for (const applicationModel of this.models) {
+          if (this.isModelSame(applicationModel)) {
+            continue
+          } else if (this.isModelEmpty(applicationModel) || applicationModel.toDelete) {
+            await ApplicationService.deleteSupplementaryApplication(applicationModel.supplementaryApplicationId)
+            delete applicationModel.supplementaryApplicationId
+            continue
+          }
+
+          const payload = {
+            ...applicationModel,
+            applicationId: this.applicationId,
+          }
+
+          if (payload.monthlyLease) {
+            payload.monthlyLease = Number(payload.monthlyLease)
+          }
+
+          if (applicationModel.supplementaryApplicationId) {
+            await ApplicationService.updateSupplementaryApplication(applicationModel.supplementaryApplicationId, payload)
+          } else {
+            const response = await ApplicationService.createSupplementaryApplication(payload)
+            applicationModel.supplementaryApplicationId = response.supplementaryApplicationId
+          }
+
+          if (this.documentsToDelete.length > 0) {
+            for (const documentID of this.documentsToDelete) {
+              await DocumentService.deleteDocument(documentID)
+            }
+            this.documentsToDelete = []
+          }
+          if (applicationModel.documentsToUpload) {
+            try {
+              await DocumentService.createDocuments(applicationModel.documentsToUpload, applicationModel.supplementaryApplicationId)
+            } catch (error) {
+              this.setFailureAlert('Failed to upload files')
+            }
+          }
+        }
+        await this.loadData()
+        this.clonedModels = cloneDeep(this.models)
+        if (showAlert) {
+          this.setSuccessAlert(`Application Saved`)
+        }
+      } catch (error) {
+        this.setFailureAlert('Failed to save supplementary applications')
+        this.loading = false
+      }
```

Comment:
we may want to add this code so that when the user click next, the AppNavButtons will not be stuck at processing state
```
 finally {
        this.$emit('process', false)
      }
```

Reply:
done